File: Accesa/Accesa.SportsBuddy/Accesa.SportsBuddy/Accesa.SportsBuddy/Database/Models/Event.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Accesa.SportsBuddy.Database.Models
{
    public partial class Event
    {
        public Event()
        {
            JoinEvents = new HashSet<JoinEvent>();
        }

        public int Id { get; set; }
        public int AuthorId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Goal { get; set; }
        public int AddressId { get; set; }
        public DateTime StartDate { get; set; }
        public string Duration { get; set; }

        public virtual Address Address { get; set; }
        public virtual User Author { get; set; }
        public virtual ICollection<JoinEvent> JoinEvents { get; set; }
    }
}

File: Accesa/Accesa.SportsBuddy/Accesa.SportsBuddy.UI/src/models/Event.ts
import { DEFAULT_TRAINEE, Trainee } from "./Trainee";
import { Address } from "./UserInfo";
import {Admin} from './Admin';

export interface Event {
    id: number;
    authorId: number;
    title: string;
    description: string;
    goal: string;
    addressId: number;
    address: Address;
    startDate: Date;
    duration: string;
    author: Trainee | Admin;
};

export const DEFAULT_EVENT = {
    id: 0,
    authorId: 1,
    title: '',
    description: '',
    goal: '',
    addressId: 0,
    address: {
        id: 0,
        street: '',
        city: '',
        state: '',
        postalCode: '',
        country: '',
    },
    author: DEFAULT_TRAINEE,
    startDate: new Date(),
    duration: ''
}

export interface JoinEvent {
    eventId: number;
    userId: number;
}

export interface EventExtended {
    event: Event;
    eventId: number;
    userId : number;
    user: Trainee;
}